addons/web/static/tests/core/confirmation_dialog_tests.js

```diff
@@ -44,21 +44,52 @@ QUnit.module("Components", (hooks) => {
                 close: () => {
                     assert.step("Close action");
                 },
-                confirm: () => {},
+                confirm: () => {
+                    throw new Error("should not be called");
+                },
                 cancel: () => {
-                    assert.step("Cancel action");
+                    throw new Error("should not be called");
+                },
+                dismiss: () => {
+                    assert.step("Dismiss action");
                 },
             },
         });
         assert.verifySteps([]);
         triggerHotkey("escape");
         await nextTick();
         assert.verifySteps(
-            ["Cancel action", "Close action"],
+            ["Dismiss action", "Close action"],
             "dialog has called its cancel method before its closure"
         );
     });
 
+    QUnit.test("clicking on 'X' to close the dialog", async function (assert) {
```

Comment:
We should probably duplicate those 2 (the escape and the clicking on the cross icon) and avoid specifying a dismiss method, and make sure it calls the cancel method in that case.